lib/createExplorer.js

```diff
@@ -77,7 +80,13 @@ module.exports = function (options) {
           ? splitPath.slice(0, -1).join(path.sep)
           : null;
 
-        if (!nextDirectory || directory === options.stopDir) return null;
+        if (!nextDirectory || directory === options.stopDir || lastSearchDirectory) {
+          if (mustEnsureUserHome && directory !== oshomedir()) {
+            return searchDirectory(oshomedir(), true);
```

Comment:
Whatever the logic ends up being, can we avoid calling this `oshomedir()` function multiple times, and instead save the value in a variable?